### ISSUE TITLE
haxe js - curly braces of `if` followed by `switch (enum)` not kept

### ISSUE THREAD

**Issue by zabojad** [HaxeFoundation/haxe#5829]
I've just spotted a bug with haxe generated javascript in a particular case.

When a `if` follows a `switch` on an enum and that same `if` has an `else` statement, the `else` statement gets "linked" to another `if` generated by the `switch` on the enum.

Below the details of my case:

The Haxe code:
```
		if (payload != null) {

			switch (payload) {

				case Json(d):

					req.setRequestHeader("Content-Type", "application/json");

					req.send(haxe.Json.stringify(d));

				case FormData(fd):

					req.setRequestHeader("Content-Type", "multipart/form-data");

					req.send(fd);
			}

		} else {

			req.send();
		}
```

The js output:
```
if(payload != null) if(payload != null) switch(payload[1]) {
	case 0:
		var d = payload[2];
		req.setRequestHeader("Content-Type","application/json");
		req.send(JSON.stringify(d));
		break;
	case 1:
		var fd = payload[2];
		req.setRequestHeader("Content-Type","multipart/form-data");
		req.send(fd);
		break;
	} else req.send();
```

The switch on my enum seems to generate a `if (%my enum% == null)` but the curly braces of the original js code are not kept which lead to a bug in case `%my enum% is null. In that case, we never get to the `else` block as we espect reading at the haxe code.

I'm having this bug with haxe 3.2.1. I haven't tested with other Haxe versions...

**Comment by Simn** [comment]
Please do test with a more recent Haxe version, I think this has been fixed.

**Comment by zabojad** [comment]
@Simn indeed, problem solved in 3.3.0 rc1
